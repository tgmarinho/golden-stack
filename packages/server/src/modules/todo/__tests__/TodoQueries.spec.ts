import { graphql } from 'graphql';
import { toGlobalId } from 'graphql-relay';

import { schema } from '../../../graphql/schema';

import {
  clearDbAndRestartCounters,
  connectMongoose,
  createTodo,
  disconnectMongoose,
  getContext,
  sanitizeTestObject,
} from '../../../../test/helpers';

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

describe('TodoType queries', () => {
  it('should query an Todo', async () => {
    const todo = await createTodo();

    // language=GraphQL
    const query = `
      query Q($id: ID) {
        todo: node(id: $id) {
          id
          ... on Todo {
            id
            title
            description
          }
        }
      }
    `;

    const variables = {
      id: toGlobalId('Todo', todo._id),
    };
    const rootValue = {};
    const context = await getContext();
    const result = await graphql(schema, query, rootValue, context, variables);

    expect(result.errors).toBeUndefined();
    expect(sanitizeTestObject(result)).toMatchSnapshot();
  });

  it('should query all Todos', async () => {
    await createTodo();
    await createTodo();
    await createTodo();
    await createTodo();
    await createTodo();

    // language=GraphQL
    const query = `
      query Q {
        todos(first: 10) {
          edges {
            node {
              id
              title
              description
            }
          }
        }
      }
    `;

    const variables = {};
    const rootValue = {};
    const context = await getContext();
    const result = await graphql(schema, query, rootValue, context, variables);

    expect(result.errors).toBeUndefined();
    expect(sanitizeTestObject(result)).toMatchSnapshot();
  });

  it('should search all Todos', async () => {
    await createTodo({ description: 'desc one' });
    await createTodo({ title: 'title one' });
    await createTodo({ description: 'three' });
    await createTodo({ description: 'two' });
    await createTodo({ title: 'title two' });

    // language=GraphQL
    const query = `
      query Q {
        todos(first: 10, search: "two") {
          edges {
            node {
              id
              title
              description
            }
          }
        }
      }
    `;

    const variables = {};
    const rootValue = {};
    const context = await getContext();
    const result = await graphql(schema, query, rootValue, context, variables);

    expect(result.errors).toBeUndefined();
    expect(sanitizeTestObject(result)).toMatchSnapshot();
  });
});
