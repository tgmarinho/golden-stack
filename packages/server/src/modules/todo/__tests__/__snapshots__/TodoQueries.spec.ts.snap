// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoType queries should query all Todos 1`] = `
Object {
  "data": Object {
    "todos": Object {
      "edges": Array [
        Object {
          "node": Object {
            "description": "This is an awesome todo #1",
            "id": "FROZEN-ID",
            "title": "Todo #1",
          },
        },
        Object {
          "node": Object {
            "description": "This is an awesome todo #2",
            "id": "FROZEN-ID",
            "title": "Todo #2",
          },
        },
        Object {
          "node": Object {
            "description": "This is an awesome todo #3",
            "id": "FROZEN-ID",
            "title": "Todo #3",
          },
        },
        Object {
          "node": Object {
            "description": "This is an awesome todo #4",
            "id": "FROZEN-ID",
            "title": "Todo #4",
          },
        },
        Object {
          "node": Object {
            "description": "This is an awesome todo #5",
            "id": "FROZEN-ID",
            "title": "Todo #5",
          },
        },
      ],
    },
  },
}
`;

exports[`TodoType queries should query an Todo 1`] = `
Object {
  "data": Object {
    "todo": Object {
      "description": "This is an awesome todo #NaN",
      "id": "FROZEN-ID",
      "title": "Todo #NaN",
    },
  },
}
`;

exports[`TodoType queries should search all Todos 1`] = `
Object {
  "data": Object {
    "todos": Object {
      "edges": Array [
        Object {
          "node": Object {
            "description": "two",
            "id": "FROZEN-ID",
            "title": "Todo #4",
          },
        },
        Object {
          "node": Object {
            "description": "This is an awesome todo #5",
            "id": "FROZEN-ID",
            "title": "title two",
          },
        },
      ],
    },
  },
}
`;
